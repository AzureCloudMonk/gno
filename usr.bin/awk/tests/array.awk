# Read values into an array

$1 == "REMOVE" { delete value[$2]; next }

$1 == "RESET" {delete value; next}

$1 == "SKIP" {nextfile}

$1 == "PRINT" {printarray(value); next}

{ value[$1] = $2 }

END { printarray(value) }

function printarray(A,  key) {
  print "Values in array";
  for (key in A) print "  ",key,A[key];
  print " ";
}
