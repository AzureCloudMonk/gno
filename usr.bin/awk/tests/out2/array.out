Values in array
   nine 9
   two 2
   one 1
   three 3
   six 6
   four 4
   seven 7
   five 5
   eight 8
 
Values in array
   two 2
   one 1
   three 3
   six 6
   four 4
   seven 7
   five 5
   eight 8
 
Values in array
   two II
   one I
   three III
   six VI
   four IV
   five V
 
